src/SDK/Subscriber/Dispatcher.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Subscriber;
+
+class Dispatcher
+{
+    private static array $Listener = [];
+    private static $instance = null;
+
+    private function __construct()
+    {
+    }
+
+    public static function getInstance()
+    {
+        if (self::$instance == null) {
+            self::$instance = new Dispatcher();
+        }
+
+        return self::$instance;
+    }
+
+    public function setListener(string $event, string $listener):void
+    {
+        self::$Listener[$event] = $listener;
+    }
+
+    public function dispatch(EventInterface $event):void
+    {
+        $eventClassname = get_class($event);
+        $eventClassname = str_replace('OpenTelemetry\\SDK\\Subscriber\\Event\\', '', $eventClassname);
```

Comment:
I think it's better if we use `$event::class` as the key, so there can be no ambiguity between classes in different namespaces?

Reply:
The `class` constant is not available on objects in php 7.4, so it should simply be `get_class` or the event class should have a `name` parameter/method.